### DISCUSSION TITLE
Crashing on OS 12.6 / M2 mac, "Namespace DYLD, Code 1 Library missing"

### DISCUSSION THREAD

**Discussion by mikeyvids** [vgmtrans/vgmtrans#347] [Q&A] [answered]
I've installed this on MacBook Air, OS 12.6, Apple Silicon M2 and it crashes each attempt when opening.

Any suggestions on anything I might be missing or an alternative way of installing? 

It gave the error "Namespace DYLD, Code 1 Library missing". After a little digging I installed an updated version of brew, based on suggestion. However the issue persisted. I copied the report after the crash as follows:

-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               VGMTrans [995]
Path:                  /Users/USER/Desktop/VGMTrans.app/Contents/MacOS/VGMTrans
Identifier:            org.vgmtrans.vgmtrans
Version:               ???
Code Type:             X86-64 (Translated)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2023-03-07 11:47:46.7780 -0500
OS Version:            macOS 12.6 (21G115)
Report Version:        12
Anonymous UUID:        DC372AD1-E2A5-104F-654D-51D36BA8775E

Time Awake Since Boot: 110 seconds

System Integrity Protection: disabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace DYLD, Code 1 Library missing

(terminated at launch; ignore backtrace)

Kernel Triage:
VM - pmap_enter failed with resource shortage

Thread 0 Crashed:
0   Rosetta Runtime Routines	       0x104f27940 ???
1   <translation info unavailable>	       0x104f7a5ac ???
2   dyld                          	       0x204f79a72 abort_with_payload_wrapper_internal + 80
3   dyld                          	       0x204f79aa4 abort_with_payload + 9
4   dyld                          	       0x204f27699 dyld4::halt(char const*) + 615
5   dyld                          	       0x204f235b5 dyld4::prepare(dyld4::APIs&, dyld3::MachOAnalyzer const*) + 3995
6   dyld                          	       0x204f224e4 start + 388

Thread 1:: com.apple.rosetta.exceptionserver
0   runtime                       	    0x7ff7fff22944 0x7ff7fff1e000 + 18756
1   runtime                       	    0x7ff7fff303e0 0x7ff7fff1e000 + 74720
2   runtime                       	    0x7ff7fff31e58 0x7ff7fff1e000 + 81496

Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000006   x1: 0x0000000000000001   x2: 0x000000030d7dd240   x3: 0x0000000000000098
    x4: 0x000000030d7dce40   x5: 0x0000000000000000   x6: 0x00007ff80bb3f000   x7: 0x000000208fb0c71c
    x8: 0x000000030d7dce40   x9: 0x0000000000000000  x10: 0x0000000000000098  x11: 0x000000030d7dd27e
   x12: 0x0000000000000098  x13: 0x000000030d7dd240  x14: 0x0000000000000001  x15: 0x0000000000000006
   x16: 0x0000000000000209  x17: 0x0000000000000000  x18: 0x800000010cf90b70  x19: 0x0000000104f1d000
   x20: 0x00000000000007ff  x21: 0x000000010cf95fa0  x22: 0xffffffffffffffff  x23: 0x0000000104f27914
   x24: 0x0000000204f79a72  x25: 0x0000000104f271f4  x26: 0x0000000000000000  x27: 0x0000000000000000
   x28: 0x0000000204f6398e   fp: 0x000000002c000000   lr: 0x0000000104f7a5ac
    sp: 0x000000010cf90b70   pc: 0x0000000104f27940 cpsr: 0x80001000
   far: 0x0000000104f7a000  esr: 0x56000080  Address size fault

Binary Images:
               0x0 - 0xffffffffffffffff ??? (*) <00000000-0000-0000-0000-000000000000> ???
       0x204f1d000 -        0x204f88fff dyld (*) <71febccd-d9dc-3599-9971-2b3407c588a8> /usr/lib/dyld
    0x7ff7fff1e000 -     0x7ff7fff4dfff runtime (*) <1f0f0114-cfb6-304b-8183-8ef52c21f8b4> /usr/libexec/rosetta/runtime

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 2
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=944.7M resident=0K(0%) swapped_out_or_unallocated=944.7M(100%)
Writable regions: Total=156.0M written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=156.0M(100%)

                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Rosetta Arena                     4096K        2 
Rosetta Generic                    792K      195 
Rosetta IndirectBranch              32K        1 
Rosetta JIT                      128.0M        1 
Rosetta Return Stack                20K        2 
Rosetta Thread Context              20K        2 
Stack                             8176K        1 
Stack Guard                       56.0M        1 
VM_ALLOCATE                       13.3M        6 
VM_ALLOCATE (reserved)             232K       14         reserved VM address space (unallocated)
__DATA                            16.1M      260 
__DATA_CONST                      11.6M      161 
__DATA_DIRTY                       485K       85 
__FONT_DATA                          4K        1 
__LINKEDIT                       708.8M       25 
__OBJC_RO                         82.8M        1 
__OBJC_RW                         3200K        1 
__TEXT                           236.0M      275 
__UNICODE                          592K        1 
dyld private memory               1024K        1 
mapped file                        5.1G      356 
===========                     =======  ======= 
TOTAL                              6.4G     1392 
TOTAL, minus reserved VM space     6.4G     1392 

-----------
Full Report
-----------

{"app_name":"VGMTrans","timestamp":"2023-03-07 11:47:49.00 -0500","app_version":"","slice_uuid":"aaa28ce0-9d3c-358c-a79e-1bb4dcb4bdd4","build_version":"","platform":1,"bundleID":"org.vgmtrans.vgmtrans","share_with_app_devs":1,"is_first_party":0,"bug_type":"309","os_version":"macOS 12.6 (21G115)","incident_id":"D9682CD3-D8A9-4CC5-AC9C-0BFD5CBD070D","name":"VGMTrans"}
{
  "uptime" : 110,
  "procLaunch" : "2023-03-07 11:47:46.6105 -0500",
  "procRole" : "Default",
  "version" : 2,
  "userID" : 501,
  "deployVersion" : 210,
  "modelCode" : "Mac14,2",
  "procStartAbsTime" : 2770888280,
  "coalitionID" : 1021,
  "osVersion" : {
    "train" : "macOS 12.6",
    "build" : "21G115",
    "releaseType" : "User"
  },
  "captureTime" : "2023-03-07 11:47:46.7780 -0500",
  "incident" : "D9682CD3-D8A9-4CC5-AC9C-0BFD5CBD070D",
  "bug_type" : "309",
  "pid" : 995,
  "procExitAbsTime" : 2771649116,
  "translated" : true,
  "cpuType" : "X86-64",
  "procName" : "VGMTrans",
  "procPath" : "\/Users\/USER\/Desktop\/VGMTrans.app\/Contents\/MacOS\/VGMTrans",
  "bundleInfo" : {"CFBundleIdentifier":"org.vgmtrans.vgmtrans"},
  "storeInfo" : {"deviceIdentifierForVendor":"059CEC2C-1B24-5CA9-B8A5-CC6E0A2A1346","thirdParty":true},
  "parentProc" : "launchd",
  "parentPid" : 1,
  "coalitionName" : "org.vgmtrans.vgmtrans",
  "crashReporterKey" : "DC372AD1-E2A5-104F-654D-51D36BA8775E",
  "sip" : "disabled",
  "isCorpse" : 1,
  "exception" : {"codes":"0x0000000000000000, 0x0000000000000000","rawCodes":[0,0],"type":"EXC_CRASH","signal":"SIGABRT"},
  "termination" : {"flags":518,"code":1,"namespace":"DYLD","indicator":"Library missing","details":["(terminated at launch; ignore backtrace)"],"reasons":[]},
  "ktriageinfo" : "VM - pmap_enter failed with resource shortage\n",
  "extMods" : {"caller":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"system":{"thread_create":0,"thread_set_state":0,"task_for_pid":2},"targeted":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"warnings":0},
  "faultingThread" : 0,
  "threads" : [{"triggered":true,"id":9126,"threadState":{"x":[{"value":6},{"value":1},{"value":13111251520},{"value":152},{"value":13111250496},{"value":0},{"value":140703324958720},{"value":139849680668},{"value":13111250496},{"value":0},{"value":152},{"value":13111251582},{"value":152},{"value":13111251520},{"value":1},{"value":6},{"value":521},{"value":0},{"value":9223372041367391088},{"value":4377923584},{"value":2047},{"value":4512636832},{"value":18446744073709551615},{"value":4377966868},{"value":8673270386,"symbolLocation":80,"symbol":"abort_with_payload_wrapper_internal"},{"value":4377965044},{"value":0},{"value":0},{"value":8673180046,"symbolLocation":10,"symbol":"__abort_with_payload"}],"flavor":"ARM_THREAD_STATE64","lr":{"value":4378305964},"cpsr":{"value":2147487744},"fp":{"value":738197504},"sp":{"value":4512615280},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":4377966912,"matchesCrashFrame":1},"far":{"value":4378304512}},"frames":[{"imageOffset":4377966912,"region":"Rosetta Runtime Routines","symbolLocation":10560,"imageIndex":0},{"imageOffset":4378305964,"region":"<translation info unavailable>","imageIndex":0},{"imageOffset":379506,"symbol":"abort_with_payload_wrapper_internal","symbolLocation":80,"imageIndex":1},{"imageOffset":379556,"symbol":"abort_with_payload","symbolLocation":9,"imageIndex":1},{"imageOffset":42649,"symbol":"dyld4::halt(char const*)","symbolLocation":615,"imageIndex":1},{"imageOffset":26037,"symbol":"dyld4::prepare(dyld4::APIs&, dyld3::MachOAnalyzer const*)","symbolLocation":3995,"imageIndex":1},{"imageOffset":21732,"symbol":"start","symbolLocation":388,"imageIndex":1}]},{"id":9128,"name":"com.apple.rosetta.exceptionserver","frames":[{"imageOffset":18756,"imageIndex":2},{"imageOffset":74720,"imageIndex":2},{"imageOffset":81496,"imageIndex":2}]}],
  "usedImages" : [
  {
    "size" : 0,
    "source" : "A",
    "base" : 0,
    "uuid" : "00000000-0000-0000-0000-000000000000"
  },
  {
    "source" : "P",
    "arch" : "x86_64",
    "base" : 8672890880,
    "size" : 442368,
    "uuid" : "71febccd-d9dc-3599-9971-2b3407c588a8",
    "path" : "\/usr\/lib\/dyld",
    "name" : "dyld"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 140703127691264,
    "size" : 196608,
    "uuid" : "1f0f0114-cfb6-304b-8183-8ef52c21f8b4",
    "path" : "\/usr\/libexec\/rosetta\/runtime",
    "name" : "runtime"
  }
],
  "sharedCache" : {
  "base" : 140703210061824,
  "size" : 19331678208,
  "uuid" : "3765e290-e8ae-3af7-92bf-da62dfb8ac4f"
},
  "vmSummary" : "ReadOnly portion of Libraries: Total=944.7M resident=0K(0%) swapped_out_or_unallocated=944.7M(100%)\nWritable regions: Total=156.0M written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=156.0M(100%)\n\n                                VIRTUAL   REGION \nREGION TYPE                        SIZE    COUNT (non-coalesced) \n===========                     =======  ======= \nRosetta Arena                     4096K        2 \nRosetta Generic                    792K      195 \nRosetta IndirectBranch              32K        1 \nRosetta JIT                      128.0M        1 \nRosetta Return Stack                20K        2 \nRosetta Thread Context              20K        2 \nStack                             8176K        1 \nStack Guard                       56.0M        1 \nVM_ALLOCATE                       13.3M        6 \nVM_ALLOCATE (reserved)             232K       14         reserved VM address space (unallocated)\n__DATA                            16.1M      260 \n__DATA_CONST                      11.6M      161 \n__DATA_DIRTY                       485K       85 \n__FONT_DATA                          4K        1 \n__LINKEDIT                       708.8M       25 \n__OBJC_RO                         82.8M        1 \n__OBJC_RW                         3200K        1 \n__TEXT                           236.0M      275 \n__UNICODE                          592K        1 \ndyld private memory               1024K        1 \nmapped file                        5.1G      356 \n===========                     =======  ======= \nTOTAL                              6.4G     1392 \nTOTAL, minus reserved VM space     6.4G     1392 \n",
  "legacyInfo" : {
  "threadTriggered" : {

  }
},
  "trialInfo" : {
  "rollouts" : [
    {
      "rolloutId" : "61af99aeda72d16a4beb7756",
      "factorPackIds" : {

      },
      "deploymentId" : 240000409
    },
    {
      "rolloutId" : "60356660bbe37970735c5624",
      "factorPackIds" : {

      },
      "deploymentId" : 240000027
    }
  ],
  "experiments" : [

  ]
}
}

Model: Mac14,2, BootROM 7459.141.1, proc 8:4:4 processors, 24 GB, SMC 
Graphics: Apple M2, Apple M2, Built-In
Display: LG HDR 4K, 3840 x 2160 (2160p/4K UHD 1 - Ultra High Definition), Main, MirrorOff, Online
Display: Color LCD, spdisplays_2560x1664Retina, MirrorOff, Online
Memory Module: LPDDR5
AirPort: Wi-Fi, wl0: Apr  6 2022 05:55:54 version 20.90.45.0.8.7.118 FWID 01-e7138ff2
Bluetooth: Version (null), 0 services, 0 devices, 0 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
USB Device: USB31Bus
USB Device: USB31Bus
USB Device: USB2.1 Hub
USB Device: LG Monitor Controls
Thunderbolt Bus: MacBook Air, Apple Inc.
Thunderbolt Bus: MacBook Air, Apple Inc.

**Comment by sykhro** [comment, marked as answer]
Seems like this has broken after we updated to Qt 6. I have just reverted it back to Qt 5, please try to run the new build once it's compiled (it will be linked [at the bottom of this page](https://github.com/vgmtrans/vgmtrans/actions/runs/4356627047) once green ticks appear)

**Reply by mikeyvids** [reply]
Yes, that worked. You made my day! Thank you

**Comment by mikeyvids** [comment]
Yes, that worked. Made my day!

Thank you

**Comment by sykhro** [comment]
No, thank you for taking the time to report this :)
